### PR TITLE
Voxels fixes

### PR DESCRIPTION
**Pull request by jjhembd**
This PR addresses 3 issues:
- Shader compilation errors on Windows
- Test failures
- `build-ts` errors (due to problematic doc strings)

I would normally address these in separate PRs, but I wasn't sure how to verify my changes until I could get CI to pass.

The most significant code changes include:
- `VoxelFS.glsl` is broken down into 13 separate files. 
- `VoxelDrawCommands` is pulled out of `VoxelPrimitive`, and updated to add only the needed shader files, depending on the voxel shape
- `getOctreeLeafSampleData` (in `Octree.glsl`) has alternate definitions based on the value of `SAMPLE_COUNT`. This fixes the Windows linking error described in https://github.com/CesiumGS/cesium/pull/10253#issuecomment-1216759961.
- The classes `KeyframeNode`, `SpatialNode`, and `Megatexture` are pulled out of `VoxelTraversal` for separate documentation and testing (this PR only adds `KeyframeNodeSpec`)

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @jjhembd!
* :heavy_check_mark: Signed CLA found.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Comment by lilleyse** [comment]
Just minor things from me. Looks good @jjhembd

File: Source/Scene/VoxelBoxShape.js [hunk 1]
```diff
@@ -219,13 +228,6 @@ VoxelBoxShape.prototype.update = function (
       (renderMinBounds.y === renderMaxBounds.y) +
       (renderMinBounds.z === renderMaxBounds.z) >=
       2 ||
-    clipMinBounds.x > clipMaxBounds.x ||
-    clipMinBounds.y > clipMaxBounds.y ||
-    clipMinBounds.z > clipMaxBounds.z ||
-    (clipMinBounds.x === clipMaxBounds.x) +
-      (clipMinBounds.y === clipMaxBounds.y) +
-      (clipMinBounds.z === clipMaxBounds.z) >=
-      2 ||
```
[the remark this answers is not in the document]
Comment (jjhembd) [reply Source/Scene/VoxelBoxShape.js:228]: `renderMinBounds` and `renderMaxBounds` were already clamped to be between `clipMinBounds` and `clipMaxBounds`. So `clipMinBounds.x === clipMaxBounds.x` is equivalent to the previous check `renderMinBounds.x === renderMaxBounds.x`.

I thought I had a similar argument for `clipMinBounds > clipMaxBounds`, but now I'm not so sure. I'll add those checks back, and also clarify the comment.

File: Source/Shaders/Voxels/VoxelFS.glsl [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    uniform vec4 u_pickColor;
+#endif
+
+#if defined(JITTER)
+float hash(vec2 p)
+{
+	vec3 p3 = fract(vec3(p.xyx) * 50.0); // magic number = hashscale
+	p3 += dot(p3, p3.yzx + 19.19);
+	return fract((p3.x + p3.y) * p3.z);
+}
```
[the remark this answers is not in the document]
Comment (jjhembd) [reply Source/Shaders/Voxels/VoxelFS.glsl]: Thanks, I fixed these three. There are a few more in `FilmicToneMapping.glsl`, but I'll leave those for a separate PR.

**Comment by jjhembd** [comment]
Updated, thanks @lilleyse!

**Pull request merged**